src/main/kotlin/com/college/hotlittlepigs/user/UserResource.kt

```diff
@@ -0,0 +1,101 @@
+package com.college.hotlittlepigs.user
+
+import com.college.hotlittlepigs.log.Log
+import com.college.hotlittlepigs.log.LogService
+import com.college.hotlittlepigs.pagination.PageModel
+import com.college.hotlittlepigs.pagination.PageRequestModel
+import com.college.hotlittlepigs.security.JwtManager
+import com.college.hotlittlepigs.user.dto.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.http.ResponseEntity.ok
+import org.springframework.http.ResponseEntity.status
+import org.springframework.security.access.annotation.Secured
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.web.bind.annotation.*
+import javax.validation.Valid
+import org.springframework.security.core.userdetails.User as SpringUser
+
+@RequestMapping("users")
+@RestController
+class UserResource {
+    @Autowired
+    private lateinit var userService: UserService
+
+    @Autowired
+    private lateinit var logService: LogService
+
+    @Autowired
+    private lateinit var jwtManager: JwtManager
+
+    @Autowired
+    private lateinit var authManager: AuthenticationManager
+
+    @PostMapping
+    fun save(@RequestBody @Valid userDTO: UserSaveDTO): ResponseEntity<User> {
+        val user = userDTO.toUser()
+        val createdUser = userService.save(user)
+        return status(HttpStatus.CREATED).body(createdUser)
+    }
+
+    @PreAuthorize("@accessManager.isOwner(#id)")
+    @PutMapping("/{id}")
+    fun update(@PathVariable id: Long, @RequestBody userDTO: UserUpdateDTO): ResponseEntity<User> {
+        val user = userDTO.toUser()
+        user.id = id
+        // TODO we should search by id, not set the id
+        val updatedUser = userService.save(user)
```

Comment:
Aqui eu coloquei errado
```suggestion
        val updatedUser = userService.update(user)
```